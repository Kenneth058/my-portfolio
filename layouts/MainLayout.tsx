import * as React from 'react'
import Head from 'next/head'

export interface Props {
  children: React.ReactNode
}

const MainLayout: React.FunctionComponent<Props> = ({ children }) => (
  <div className="bg-black-1 min-h-screen">
    <Head>
      <title>Kenneth Ch√©vez</title>
      <meta name="viewport" content="initial-scale=1.0, width=device-width" />
    </Head>
    {children}
  </div>
)

export default MainLayout
