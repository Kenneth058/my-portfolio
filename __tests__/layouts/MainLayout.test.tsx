import * as React from 'react'
import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import MainLayout from '../../layouts/MainLayout'

describe('MainLayout', () => {
  it('renders child', () => {
    render(<MainLayout>Main Child</MainLayout>)

    const text = screen.getByText(/main child/i)
    expect(text).toBeInTheDocument()
  })

  it('renders child', () => {
    render(
      <MainLayout>
        <p>Other text</p>
      </MainLayout>
    )

    const text = screen.getByText(/other text/i)
    expect(text).toBeInTheDocument()
  })

  it('renders MainLayout unchanged', () => {
    const tree = renderer.create(<MainLayout>Main Child</MainLayout>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
