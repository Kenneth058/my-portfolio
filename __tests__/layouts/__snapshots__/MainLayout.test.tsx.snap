// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout renders MainLayout unchanged 1`] = `
<div
  className="bg-black-1 min-h-screen"
>
  Main Child
</div>
`;
